libraries/platform/iot_network.h

```diff
@@ -356,6 +356,10 @@ struct IotNetworkCredentials
     size_t clientCertSize;    /**< @brief Size associated with #IotNetworkCredentials.pClientCert. */
     const char * pPrivateKey; /**< @brief String representing the client certificate's private key. */
     size_t privateKeySize;    /**< @brief Size associated with #IotNetworkCredentials.pPrivateKey. */
+    const char * pUserName;   /**< @brief String representing the username for MQTT. */
+    size_t userNameSize;      /**< @brief Size associated with #IotNetworkCredentials.pUserName. */
+    const char * pPassword;   /**< @brief String representing the password for MQTT. */
+    size_t passwordSize;      /**< @brief Size associated with #IotNetworkCredentials.pPassword. */
```

Comment:
Is there a case in which both credentials and user/name password will be used? If not can we make this a union instead?

Reply:
Yes, the settings are orthogonal. The web gateway may or may not require client certificate authentication, and the MQTT broker may or may not require username/password.

Reply:
ok. thanks